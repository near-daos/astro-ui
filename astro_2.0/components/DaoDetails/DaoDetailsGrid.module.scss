@import 'styles/typography';
@import 'styles/mixins';

.root {
  @include card-shadow;
  height: 440px;
  max-width: 458px;
  min-width: 270px;
  transition: box-shadow 0.4s;
  width: 100%;

  &:hover {
    box-shadow: 3px 2px 24px var(--color-black-opacity-25);
  }
}

.content {
  display: flex;
  flex-direction: column;
  height: 100%;
  justify-content: space-between;
  padding-bottom: 32px;
  width: 100%;
}

.general {
  align-items: center;
  display: flex;
}

.flag {
  flex: 0 0 69px;
  height: 78px;
  margin: 8px 10px;
  position: relative;
  width: 69px;

  @include tablet-portrait {
    margin: 20px 24px;
  }

  @include tablet-landscape {
    margin: 28px 24px;
  }

  .background {
    background-color: var(--color-white);
    clip-path: url('#flag');
    inset: 0;
    position: absolute;
    transform: scale(1.04, 1.045);
  }

  .cover {
    background-position: center;
    background-repeat: no-repeat;
    background-size: cover;
    clip-path: url('#flag');
    height: 100%;
    width: 100%;
  }

  .cover.scaled {
    transform: scale(1.45);
  }
}

.title {
  padding: 12px 16px 12px 0;
  width: 100%;

  @include tablet-portrait {
    padding-right: 24px;
  }
}

.name {
  @include font(28px, 28px, 700);
  @include ellipse-text;
  font-family: 'Roboto', 'sans-serif';

  @include astro-mobile {
    font-size: 20px;
  }
}

.nameTooltip {
  display: block;
  max-width: 200px;
  white-space: normal;
}

.address {
  @include font(16px, 24px, 500);
  align-items: center;
  color: var(--color-neutral-60);
  display: flex;
  margin-top: 8px;
  word-break: break-all;

  @include astro-mobile {
    font-size: 14px;
    > .addressId,
    > .copyAddress {
      @include ellipse-text;
      max-width: calc(70%);
    }
  }
}

.copyAddress {
  > button {
    margin-left: 6px;
  }
}

.description {
  @include font(16px, 24px, 400);
  height: 72px;
  overflow: hidden;
  padding: 0 24px;
}

.fundsAndMembers {
  display: flex;
  justify-content: space-around;
  padding: 0 24px;
}

.controls {
  @include centralize;
  justify-content: space-around;
  padding: 14px;

  @media (min-width: 480px) {
    justify-content: space-between;
    padding: 14px 28px;
  }

  @include tablet-landscape {
    justify-content: space-around;
    padding: 14px;
  }

  @include astro-desktop {
    justify-content: space-between;
    padding: 14px 28px;
  }

  @media (min-width: 1150px) {
    padding: 14px 20px;
  }

  @include large-desktop {
    padding: 14px 28px;
  }
}

.controlIcon {
  @include centralize;
  border-radius: 6px;
  transition: 0.2s linear;
  width: 42px;

  @media (min-width: 480px) {
    border-radius: 8px;
    width: auto;
  }

  @include tablet-landscape {
    border-radius: 6px;
    width: 42px;
  }

  @include astro-desktop {
    border-radius: 8px;
    width: auto;
  }

  > button {
    color: var(--color-neutral-70);
    width: 100%;

    @media (min-width: 480px) {
      @include font(10px, 15px, 500);
      flex-direction: row;
      padding: 0 14px;
    }

    @include tablet-landscape {
      @include font(8px, 12px, 400);
      flex-direction: column;
      padding: 0;
    }

    @include astro-desktop {
      @include font(10px, 15px, 500);
      flex-direction: row;
      padding: 0 14px;
    }
  }

  svg {
    width: 16px;

    @media (min-width: 480px) {
      margin: 0 6px 0 0;
    }

    @include tablet-landscape {
      margin: 0 0 4px;
    }

    @include astro-desktop {
      margin: 0 6px 0 0;
    }
  }
}

.controlIcon:hover {
  background-color: var(--color-neutral-30);
  color: var(--color-brand-black);

  > button {
    color: var(--color-brand-black);
  }
}

.proposals {
  @include centralize;
  flex-direction: column;
  padding: 0 16px;
  width: 100%;
}

.proposalsTitle {
  margin-bottom: 8px;

  &.active {
    font-size: 1.375em;
    font-weight: 300;
    line-height: 1.25em;
  }

  span {
    font-weight: 700;
  }
}

.proposalsSubtitle {
  color: var(--color-neutral-60);
  margin: 0;
  white-space: nowrap;
}
