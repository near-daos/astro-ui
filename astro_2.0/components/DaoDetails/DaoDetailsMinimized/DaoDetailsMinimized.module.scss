@import 'styles/typography';
@import 'styles/mixins';

.root {
  background: var(--color-white);
  box-shadow: 3px 2px 24px rgba(0, 0, 0, 0.1);
  display: flex;
  height: 66px;
  margin: 0 -24px;
  transition: box-shadow 0.4s;

  @include tablet-landscape {
    border-radius: 8px;
    display: flex;
    margin: 0;
  }

  &:hover {
    box-shadow: 3px 2px 24px var(--color-black-opacity-25);
  }
}

.wrapper {
  align-items: center;
  display: flex;
  -ms-overflow-style: none;
  overflow-x: auto;
  overflow-y: hidden;
  scrollbar-width: none;
  width: 100%;
}

.wrapper::-webkit-scrollbar {
  display: none;
}

.paddingWithNoProposalButton {
  padding: 0 0 0 24px;
}

.paddingWithProposalButton {
  padding: 0 0 0 70px;
}

.general {
  align-items: center;
  display: flex;
  grid-area: general;

  @include tablet-landscape {
    padding: 8px 0 8px 16px;
  }

  > div:last-of-type {
    flex: 1 1 100%;
  }
}

.controls {
  align-items: center;
  display: flex;
  justify-content: space-around;
  padding: 0 5px;

  @media (min-width: 375px) {
    padding: 0 16px;
  }

  @include tablet-portrait {
    padding: 0 32px;
  }

  @include tablet-landscape {
    justify-content: flex-end;
    margin-left: auto;
    padding: 12px 0;
  }

  > div,
  a {
    @include tablet-landscape {
      margin-right: 4px;
    }

    @include desktop {
      margin-right: 6px;
    }

    @include large-desktop {
      margin-right: 8px;
    }
  }
}

.proposals {
  position: absolute;
  @include tablet-landscape {
    height: 100%;
    position: relative;
  }
}

.flagWrapper {
  flex: 0 0 47px;
  height: 53px;
  margin-right: 22px;
  width: 47px;
}

.displayName {
  @extend %title1;
  align-items: flex-end;
  display: flex;
  font-size: 18px;
  line-height: 1.1;
  max-width: 300px;
  white-space: nowrap;

  @include astro-desktop {
    max-width: 400px;
  }
}

.name {
  @include ellipse-text;
}

.explorerLink {
  margin-left: 12px;
}

.daoId {
  @extend %body2;
  color: var(--color-neutral-60);
  display: flex;
  white-space: nowrap;
}

.daoId {
  margin-right: 4px;
}

.createPopup {
  display: none;

  @include non-touch-devices {
    display: block;
  }
}

.createIcon {
  display: block;
}

.createText {
  font-size: 16px;
  font-weight: 700;
  text-transform: initial;

  @include tablet-landscape {
    display: none;
  }
}

.controlIcon {
  @include centralize;
  border-radius: 6px;
  position: relative;
  width: 42px;

  &:not(:first-of-type) {
    margin-left: 24px;
  }

  > button {
    color: var(--color-neutral-60);
    transition: 0.2s linear;
    width: 100%;
  }

  svg {
    width: 16px;
  }
}

.controlIcon.controlIcon {
  @include tablet-landscape {
    margin-left: 0;
  }
}

.controlIcon:hover {
  background-color: var(--color-neutral-30);
  color: var(--color-brand-black);

  > button {
    color: var(--color-brand-black);
  }
}

.controlIcon.active {
  background-color: var(--color-neutral-30);

  > button {
    color: var(--color-brand-black);
    cursor: default;
  }
}

.controlIcon.noActiveLink {
  color: var(--color-brand-black);

  > button {
    color: var(--color-brand-black);
  }
}
