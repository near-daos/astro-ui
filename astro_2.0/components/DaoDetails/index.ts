export * from './DaoDetailsMinimized';
export * from './DaoDetailsGrid';
