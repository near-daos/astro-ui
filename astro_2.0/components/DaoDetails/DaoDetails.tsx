import React, { FC } from 'react';
import cn from 'classnames';
import isEmpty from 'lodash/isEmpty';
import Link from 'next/link';
import { useRouter } from 'next/router';
import { useTranslation } from 'next-i18next';

import { DAO } from 'types/dao';

import { ProposalTrackerCard } from 'astro_2.0/components/DaoDetails/components/ProposalTrackerCard';
import { DaoGeneralCard } from 'astro_2.0/components/DaoDetails/components/DaoGeneralCard';
import { ActionButton } from 'features/proposal/components/action-button';
import { isUserPermittedToCreateProposal } from 'astro_2.0/features/CreateProposal/createProposalHelpers';
import { getAccumulatedTokenValue } from 'features/treasury/helpers';
import { Token } from 'types/token';

import styles from './DaoDetails.module.scss';

export interface DaoDetailsProps {
  className?: string;
  dao: DAO;
  accountId?: string | null;
  onCreateProposalClick?: (dao: DAO) => void;
  activeProposals: number;
  totalProposals: number;
  restrictCreateProposals?: boolean;
  daoTokens: Record<string, Token>;
}

export const DaoDetails: FC<DaoDetailsProps> = ({
  dao,
  className,
  daoTokens,
  accountId,
  onCreateProposalClick,
  activeProposals,
  totalProposals,
  restrictCreateProposals = false,
}) => {
  const router = useRouter();
  const { t } = useTranslation();

  const isCreateOptionAvailable = isUserPermittedToCreateProposal(
    accountId,
    dao
  );

  const action =
    !isCreateOptionAvailable ||
    restrictCreateProposals ||
    isEmpty(accountId) ? null : (
      <>Create proposal</>
    );

  function getFundsInUsdFromNear() {
    const total = getAccumulatedTokenValue(daoTokens);

    return total.toFixed(2);
  }

  return (
    <div className={cn(styles.root, className)}>
      <section className={styles.general}>
        <DaoGeneralCard
          displayName={dao.displayName}
          id={dao.id}
          description={dao.description}
          links={dao.links}
          flag={dao.logo}
          cover={dao.flagCover}
          logo={dao.flagLogo}
          legal={dao.legal}
        />
      </section>

      <section className={styles.funds}>
        <Link href={`/dao/${dao.id}/treasury/tokens`}>
          <a>
            <div className={styles.label}>{t('daoFunds')}</div>
            <div className={styles.value}>
              <span className={styles.bold}>{getFundsInUsdFromNear()}</span> USD
            </div>
          </a>
        </Link>
      </section>

      <section className={styles.members}>
        <Link href={`/dao/${dao.id}/groups/all-members`}>
          <a>
            <div className={styles.label}>{t('membersGroups')}</div>
            <div className={styles.value}>
              <span className={styles.bold}>{dao.members}</span>/
              {dao.groups.length}
            </div>
          </a>
        </Link>
      </section>

      <section className={styles.controls}>
        <ActionButton
          tooltip={t('daoSettings')}
          tooltipPlacement="top"
          iconName="settings"
          onClick={() => {
            router.push(`/dao/${dao.id}/governance/settings`);
          }}
          className={styles.controlIcon}
        />
        <ActionButton
          tooltip={t('nfts')}
          tooltipPlacement="top"
          iconName="nfts"
          onClick={() => {
            router.push(`/dao/${dao.id}/treasury/nfts`);
          }}
          className={styles.controlIcon}
        />
        <ActionButton
          tooltip={t('bounties')}
          tooltipPlacement="top"
          iconName="proposalBounty"
          onClick={() => {
            router.push(`/dao/${dao.id}/tasks/bounties`);
          }}
          className={styles.controlIcon}
        />
        <ActionButton
          tooltip={t('polls')}
          tooltipPlacement="top"
          iconName="proposalPoll"
          onClick={() => {
            router.push(`/dao/${dao.id}/tasks/polls`);
          }}
          className={styles.controlIcon}
        />
      </section>

      <section className={styles.proposals}>
        <ProposalTrackerCard
          activeVotes={activeProposals}
          totalProposals={totalProposals}
          action={action}
          onClick={() => onCreateProposalClick?.(dao)}
        />
      </section>
    </div>
  );
};
