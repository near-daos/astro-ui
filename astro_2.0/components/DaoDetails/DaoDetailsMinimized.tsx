import React, { FC, useState } from 'react';
import Link from 'next/link';
import isEmpty from 'lodash/isEmpty';
import includes from 'lodash/includes';
import { useRouter } from 'next/router';
import TextTruncate from 'react-text-truncate';

import { DAO } from 'types/dao';

import { Button } from 'components/button/Button';
import { Icon } from 'components/Icon';
import { ActionButton } from 'features/proposal/components/action-button';
import { FlagRenderer } from 'astro_2.0/components/Flag';
import { Popup } from 'components/popup/Popup';

import { useAuthContext } from 'context/AuthContext';
import { useMedia } from 'react-use';

import cn from 'classnames';
import styles from './DaoDetailsMinimized.module.scss';

export interface DaoDetailsMinimizedProps {
  dao: DAO;
  onCreateProposalClick?: () => void;
  disableNewProposal?: boolean;
}

export const DaoDetailsMinimized: FC<DaoDetailsMinimizedProps> = ({
  dao,
  onCreateProposalClick,
  disableNewProposal = false,
}) => {
  const [ref, setRef] = useState<HTMLElement | null>(null);
  const isMobile = useMedia('(max-width: 768px)');
  const tooltipPlacement = isMobile ? 'bottom' : 'top-end';

  const router = useRouter();

  const { asPath } = router;
  const currentPath = asPath.split('?')[0];

  const url = {
    funds: `/dao/${dao.id}/treasury/tokens`,
    members: `/dao/${dao.id}/groups/all-members`,
    settings: `/dao/${dao.id}/governance/settings`,
    nfts: `/dao/${dao.id}/treasury/nfts`,
    bounties: `/dao/${dao.id}/tasks/bounties`,
    polls: `/dao/${dao.id}/tasks/polls`,
  };

  const activeLinkPresent = includes(url, currentPath);

  const handleChapterClick = (newUrl: string) => {
    if (newUrl !== currentPath) {
      router.push(newUrl);
    }
  };

  const generateChapterStyle = (chapterUrl: string) => {
    return cn(styles.controlIcon, {
      [styles.active]: currentPath === chapterUrl,
      [styles.noActiveLink]: !activeLinkPresent,
    });
  };

  const { accountId, login } = useAuthContext();
  const action = (
    <>
      <div className={styles.addProposalWrapper} ref={setRef}>
        <Button
          size="block"
          onClick={() => {
            if (isEmpty(accountId)) {
              login();
            } else if (onCreateProposalClick) {
              onCreateProposalClick();
            }
          }}
          className={styles.addProposalButton}
          variant="tertiary"
        >
          <Icon width={44} name="buttonAdd" className={styles.createIcon} />
          <span className={styles.createText}>Create proposal</span>
        </Button>
      </div>
      <Popup
        offset={[0, 10]}
        anchor={ref}
        placement={tooltipPlacement}
        className={styles.createPopup}
      >
        Create proposal
      </Popup>
    </>
  );

  return (
    <div className={styles.root}>
      <Link href={`/dao/${dao.id}`}>
        <a>
          <section className={styles.general}>
            <div className={styles.flagWrapper}>
              <FlagRenderer
                flag={dao.flagCover}
                size="xs"
                fallBack={dao.logo}
              />
            </div>
            <div>
              <div className={styles.displayName}>
                <TextTruncate
                  line={1}
                  element="div"
                  truncateText="…"
                  text={dao.displayName}
                  textTruncateChild={null}
                />
              </div>
              <div className={styles.daoId}>
                <TextTruncate
                  line={1}
                  element="div"
                  truncateText="…"
                  text={dao.id}
                  textTruncateChild={null}
                />
              </div>
            </div>
          </section>
        </a>
      </Link>

      <section className={styles.controls}>
        <ActionButton
          iconName="funds"
          onClick={() => handleChapterClick(url.funds)}
          className={generateChapterStyle(url.funds)}
        >
          Funds
        </ActionButton>
        <ActionButton
          iconName="groups"
          onClick={() => handleChapterClick(url.members)}
          className={generateChapterStyle(url.members)}
        >
          Members
        </ActionButton>
        <ActionButton
          iconName="settings"
          onClick={() => handleChapterClick(url.settings)}
          className={generateChapterStyle(url.settings)}
        >
          Settings
        </ActionButton>
        <ActionButton
          iconName="nfts"
          onClick={() => handleChapterClick(url.nfts)}
          className={generateChapterStyle(url.nfts)}
        >
          NFTs
        </ActionButton>
        <ActionButton
          iconName="proposalBounty"
          onClick={() => handleChapterClick(url.bounties)}
          className={generateChapterStyle(url.bounties)}
        >
          Bounties
        </ActionButton>
        <ActionButton
          iconName="proposalPoll"
          onClick={() => handleChapterClick(url.polls)}
          className={generateChapterStyle(url.polls)}
        >
          Polls
        </ActionButton>
      </section>

      {onCreateProposalClick && !disableNewProposal && (
        <section className={styles.proposals}>{action}</section>
      )}
    </div>
  );
};
