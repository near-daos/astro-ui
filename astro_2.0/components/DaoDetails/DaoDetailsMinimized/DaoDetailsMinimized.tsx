import cn from 'classnames';
import Link from 'next/link';
import { useMedia } from 'react-use';
import includes from 'lodash/includes';
import { useRouter } from 'next/router';
import React, { FC } from 'react';
import { useTranslation } from 'next-i18next';
import { useFlags } from 'launchdarkly-react-client-sdk';

import { DAO } from 'types/dao';

import { ActionButton } from 'astro_2.0/components/ActionButton';
import { DaoLogo } from 'astro_2.0/features/DaoDashboardHeader/components/DaoLogo';
import { DaoAction } from 'astro_2.0/components/DaoDetails/DaoDetailsMinimized/components/DaoAction';
import { ExplorerLink } from 'components/ExplorerLink';
import { CopyButton } from 'astro_2.0/components/CopyButton';

import { UserPermissions } from 'types/context';
import { shortenString } from 'utils/format';

import { useDaoSettings } from 'context/DaoSettingsContext';
import { useUnreadDraftCount } from 'hooks/useDraft';
import { useWalletContext } from 'context/WalletContext';
import { useUnreadDraftProposalsCount } from 'services/ApiService/hooks/useUnreadDraftProposalsCount';

import styles from './DaoDetailsMinimized.module.scss';

export interface DaoDetailsMinimizedProps {
  dao: Pick<
    DAO,
    | 'id'
    | 'flagCover'
    | 'flagLogo'
    | 'logo'
    | 'displayName'
    | 'stakingContract'
    | 'daoMembersList'
  >;
  className?: string;
  onCreateProposalClick?: () => void;
  disableNewProposal?: boolean;
  userPermissions: UserPermissions;
}

export const DaoDetailsMinimized: FC<DaoDetailsMinimizedProps> = ({
  dao,
  className,
  onCreateProposalClick,
  userPermissions,
}) => {
  const { accountId } = useWalletContext();
  const { settings } = useDaoSettings();
  const isXsMobile = useMedia('(max-width: 600px)');
  const isMobile = useMedia('(max-width: 768px)');
  const flags = useFlags();
  const router = useRouter();
  const { t } = useTranslation();
  const { asPath } = router;
  const currentPath = asPath.split('?')[0];
  const unreadDraftsApi = useUnreadDraftCount(dao.id);
  const unreadDraftsOs = useUnreadDraftProposalsCount(dao.id);
  const unreadDrafts = unreadDraftsApi || unreadDraftsOs;

  const daoHasGovernanceTokenConfigured =
    settings?.createGovernanceToken?.wizardCompleted;

  const url = {
    delegates: `/dao/${dao.id}/delegate`,
    drafts: `/dao/${dao.id}/drafts`,
    proposals: `/dao/${dao.id}/proposals`,
    funds: `/dao/${dao.id}/treasury/tokens`,
    members: `/dao/${dao.id}/groups/all`,
    settings: `/dao/${dao.id}/governance/settings`,
    nfts: `/dao/${dao.id}/treasury/nfts`,
    bounties: `/dao/${dao.id}/tasks/bounties/list`,
    polls: `/dao/${dao.id}/tasks/polls`,
  };

  const activeLinkPresent = includes(url, currentPath);

  const generateChapterStyle = (chapter: string) => {
    return cn(styles.controlIcon, {
      [styles.active]: currentPath.indexOf(chapter) !== -1,
      [styles.noActiveLink]: !activeLinkPresent,
    });
  };

  function renderDaoLink() {
    return (
      <Link href={`/dao/${dao.id}`}>
        <a>
          <section className={cn(styles.nextPadding, styles.general)}>
            {!isMobile && (
              <div className={styles.flagWrapper}>
                <DaoLogo size="md" src={dao.flagLogo} className={styles.logo} />
              </div>
            )}

            <div>
              <div className={styles.displayName}>
                <div className={styles.name}>
                  {shortenString(dao.displayName, isXsMobile ? 25 : 40)}
                </div>

                <ExplorerLink
                  linkData={dao.id}
                  linkType="member"
                  className={styles.explorerLink}
                />
              </div>

              <div className={styles.daoId}>
                <CopyButton
                  text={dao.id}
                  tooltipPlacement="auto"
                  className={styles.copyAddress}
                >
                  <div className={styles.daoId}>
                    {shortenString(dao.id, isXsMobile ? 32 : 45)}
                  </div>
                </CopyButton>
              </div>
            </div>
          </section>
        </a>
      </Link>
    );
  }

  function renderNewProposalButton() {
    return (
      <DaoAction
        canCreateProposals={userPermissions.isCanCreateProposals}
        onCreateProposalClick={onCreateProposalClick}
        daoId={dao.id}
        dao={dao}
      />
    );
  }

  return (
    <div
      className={cn(styles.root, className, {
        [styles.nextVersion]: isMobile,
      })}
    >
      {isMobile && renderDaoLink()}
      <div className={styles.wrapper}>
        {!isMobile && renderDaoLink()}

        <section className={styles.controls}>
          {flags.delegateVoting && daoHasGovernanceTokenConfigured && (
            <ActionButton
              href={url.delegates}
              iconName="delegate"
              className={generateChapterStyle('delegates')}
            >
              {t('daoDetailsMinimized.delegates')}
            </ActionButton>
          )}
          <ActionButton
            notifications={unreadDrafts}
            href={url.drafts}
            iconName="sheet"
            className={generateChapterStyle('drafts')}
          >
            {t('daoDetailsMinimized.drafts')}
          </ActionButton>
          <ActionButton
            href={url.proposals}
            iconName="pencil"
            className={generateChapterStyle('proposals')}
          >
            {t('daoDetailsMinimized.proposals')}
          </ActionButton>

          <ActionButton
            href={url.funds}
            iconName="funds"
            className={generateChapterStyle('tokens')}
          >
            {t('daoDetailsMinimized.funds')}
          </ActionButton>

          <ActionButton
            href={url.members}
            iconName="groups"
            className={generateChapterStyle('groups')}
          >
            {t('daoDetailsMinimized.members')}
          </ActionButton>

          <ActionButton
            href={url.settings}
            iconName="settings"
            className={generateChapterStyle('settings')}
          >
            {t('daoDetailsMinimized.settings')}
          </ActionButton>

          <ActionButton
            href={url.nfts}
            iconName="nfts"
            className={generateChapterStyle('nfts')}
          >
            {t('daoDetailsMinimized.nfts')}
          </ActionButton>

          <ActionButton
            href={url.bounties}
            iconName="proposalBounty"
            className={generateChapterStyle('bounties')}
          >
            {t('daoDetailsMinimized.bounties')}
          </ActionButton>

          <ActionButton
            href={url.polls}
            iconName="proposalPoll"
            className={generateChapterStyle('polls')}
          >
            {t('daoDetailsMinimized.polls')}
          </ActionButton>
        </section>

        {!isMobile && onCreateProposalClick && !!accountId && (
          <section className={styles.proposals}>
            {renderNewProposalButton()}
          </section>
        )}
      </div>
      {isMobile && (
        <div className={styles.daoAction}>{renderNewProposalButton()}</div>
      )}
    </div>
  );
};
