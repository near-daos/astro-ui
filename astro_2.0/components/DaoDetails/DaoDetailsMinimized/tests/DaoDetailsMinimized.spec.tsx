/* eslint-disable @typescript-eslint/ban-ts-comment */
import { render, screen } from '@testing-library/react';

import { DaoDetailsMinimized } from 'astro_2.0/components/DaoDetails/DaoDetailsMinimized';

import { ProposalType } from 'types/proposal';

import { daoMock } from './mock';

jest.mock('react-text-truncate', () => {
  return () => <div />;
});

jest.mock('next/router', () => {
  return {
    useRouter: jest.fn(() => ({
      asPath: '',
    })),
  };
});

jest.mock('react-use', () => {
  return {
    ...jest.requireActual('react-use'),
    useMedia: jest.fn(),
  };
});

jest.mock('context/WalletContext', () => {
  return {
    useWalletContext: jest.fn(() => ({
      accountId: 'jasonborn.near',
    })),
  };
});

jest.mock('context/DaoSettingsContext', () => {
  return {
    useDaoSettings: jest
      .fn()
      .mockReturnValue({ settings: {}, update: jest.fn(), loading: false }),
  };
});

describe('dao details minimized', () => {
  const permissions = {
    isCanCreateProposals: true,
    isCanCreatePolicyProposals: true,
    allowedProposalsToCreate: {
      [ProposalType.ChangePolicy]: true,
      [ProposalType.ChangeConfig]: true,
      [ProposalType.AddBounty]: true,
      [ProposalType.Transfer]: true,
      [ProposalType.Vote]: true,
      [ProposalType.RemoveMemberFromRole]: true,
      [ProposalType.AddMemberToRole]: true,
      [ProposalType.AddMemberToRole]: true,
      [ProposalType.FunctionCall]: true,
      [ProposalType.UpgradeRemote]: true,
      [ProposalType.UpgradeSelf]: true,
      [ProposalType.SetStakingContract]: true,
      [ProposalType.BountyDone]: true,
    },
    allowedProposalsToVote: {
      [ProposalType.ChangePolicy]: true,
      [ProposalType.ChangeConfig]: true,
      [ProposalType.AddBounty]: true,
      [ProposalType.Transfer]: true,
      [ProposalType.Vote]: true,
      [ProposalType.RemoveMemberFromRole]: true,
      [ProposalType.AddMemberToRole]: true,
      [ProposalType.AddMemberToRole]: true,
      [ProposalType.FunctionCall]: true,
      [ProposalType.UpgradeRemote]: true,
      [ProposalType.UpgradeSelf]: true,
      [ProposalType.SetStakingContract]: true,
      [ProposalType.BountyDone]: true,
    },
  };

  afterAll(() => {
    jest.restoreAllMocks();
  });

  it('Should render component', () => {
    const { container } = render(
      <DaoDetailsMinimized dao={daoMock} userPermissions={permissions} />
    );

    expect(container).toMatchSnapshot();
  });

  it('Should render action section if onCreateProposalClick provided and user has proper permission', () => {
    render(
      <DaoDetailsMinimized
        dao={daoMock}
        userPermissions={permissions}
        onCreateProposalClick={() => 0}
      />
    );

    expect(screen.getByTestId('createProposal')).toBeInTheDocument();
    expect(screen.getByTestId('createProposal')).toBeVisible();
  });
});

/* eslint-enable @typescript-eslint/ban-ts-comment */
