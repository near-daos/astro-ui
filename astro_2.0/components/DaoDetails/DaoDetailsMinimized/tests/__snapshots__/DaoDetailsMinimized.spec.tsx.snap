// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dao details minimized Should render component 1`] = `
<div>
  <div
    class="root"
  >
    <div
      class="wrapper"
    >
      <a
        href="/dao/default-flag-test.sputnikv2.testnet"
      >
        <section
          class="nextPadding general"
        >
          <div
            class="flagWrapper"
          >
            <div
              class="root logo md"
            >
              <svg
                class="defaultLogo"
              >
                <use
                  href="/_next/undefined"
                />
              </svg>
            </div>
          </div>
          <div>
            <div
              class="displayName"
            >
              <div
                class="name"
              >
                Default Flag Test
              </div>
              <div
                class="explorerLink"
              >
                <div
                  class="root"
                  role="button"
                  tabindex="0"
                >
                  <span
                    class="iconWrapper"
                  >
                    <svg
                      class="icon"
                    >
                      <use
                        href="/_next/undefined"
                      />
                    </svg>
                  </span>
                </div>
              </div>
            </div>
            <div
              class="daoId"
            >
              <div
                class="root copyAddress"
                role="button"
                tabindex="0"
              >
                <div
                  class="daoId"
                >
                  default-flag-test.sputnikv2.testnet
                </div>
                <button
                  class="iconButton small btn"
                  type="button"
                >
                  <svg>
                    <use
                      href="/_next/undefined"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </section>
      </a>
      <section
        class="controls"
      >
        <a
          class="controlIcon noActiveLink"
          href="/dao/default-flag-test.sputnikv2.testnet/drafts"
          style="display: inline-block;"
        >
          <button
            class="iconTextButton"
            type="button"
          >
            <svg
              class="icon"
            >
              <use
                href="/_next/undefined"
              />
            </svg>
            daoDetailsMinimized.drafts
          </button>
        </a>
        <a
          class="controlIcon noActiveLink"
          href="/dao/default-flag-test.sputnikv2.testnet/proposals"
          style="display: inline-block;"
        >
          <button
            class="iconTextButton"
            type="button"
          >
            <svg
              class="icon"
            >
              <use
                href="/_next/undefined"
              />
            </svg>
            daoDetailsMinimized.proposals
          </button>
        </a>
        <a
          class="controlIcon noActiveLink"
          href="/dao/default-flag-test.sputnikv2.testnet/treasury/tokens"
          style="display: inline-block;"
        >
          <button
            class="iconTextButton"
            type="button"
          >
            <svg
              class="icon"
            >
              <use
                href="/_next/undefined"
              />
            </svg>
            daoDetailsMinimized.funds
          </button>
        </a>
        <a
          class="controlIcon noActiveLink"
          href="/dao/default-flag-test.sputnikv2.testnet/groups/all"
          style="display: inline-block;"
        >
          <button
            class="iconTextButton"
            type="button"
          >
            <svg
              class="icon"
            >
              <use
                href="/_next/undefined"
              />
            </svg>
            daoDetailsMinimized.members
          </button>
        </a>
        <a
          class="controlIcon noActiveLink"
          href="/dao/default-flag-test.sputnikv2.testnet/governance/settings"
          style="display: inline-block;"
        >
          <button
            class="iconTextButton"
            type="button"
          >
            <svg
              class="icon"
            >
              <use
                href="/_next/undefined"
              />
            </svg>
            daoDetailsMinimized.settings
          </button>
        </a>
        <a
          class="controlIcon noActiveLink"
          href="/dao/default-flag-test.sputnikv2.testnet/treasury/nfts"
          style="display: inline-block;"
        >
          <button
            class="iconTextButton"
            type="button"
          >
            <svg
              class="icon"
            >
              <use
                href="/_next/undefined"
              />
            </svg>
            daoDetailsMinimized.nfts
          </button>
        </a>
        <a
          class="controlIcon noActiveLink"
          href="/dao/default-flag-test.sputnikv2.testnet/tasks/bounties/list"
          style="display: inline-block;"
        >
          <button
            class="iconTextButton"
            type="button"
          >
            <svg
              class="icon"
            >
              <use
                href="/_next/undefined"
              />
            </svg>
            daoDetailsMinimized.bounties
          </button>
        </a>
        <a
          class="controlIcon noActiveLink"
          href="/dao/default-flag-test.sputnikv2.testnet/tasks/polls"
          style="display: inline-block;"
        >
          <button
            class="iconTextButton"
            type="button"
          >
            <svg
              class="icon"
            >
              <use
                href="/_next/undefined"
              />
            </svg>
            daoDetailsMinimized.polls
          </button>
        </a>
      </section>
    </div>
  </div>
</div>
`;
